### PR TITLE
[IDE] Pass LangOptions to `ide::isSourceInputComplete`

### LINKED ISSUE #9430: LLDB doesn't pass through language options to `ide::isSourceInputComplete`
When checking for the end of the line in the REPL, LLDB calls into the compiler's `ide::isSourceInputComplete`, but doesn't pass along any language options so the compiler doesn't know if e.g bare slash regex is enabled, resulting in the issue reported by https://forums.swift.org/t/regex-compiler-vs-repl/75393. We ought to add a language option parameter and pass along the language options for the REPL's SwiftASTContext.

### PR DESCRIPTION
**Pull request by hamishknight**
Ensure we account for things like the enablement of bare slash regex literals.

Resolves swiftlang/llvm-project#9430

### CODE REVIEW

**Comment by hamishknight** [comment]
https://github.com/swiftlang/llvm-project/pull/9439

@swift-ci please test

**Comment by hamishknight** [comment]
https://github.com/swiftlang/llvm-project/pull/9439

@swift-ci please test

File: lib/DriverTool/swift_indent_main.cpp
Comment (ahoppen) [inline lib/DriverTool/swift_indent_main.cpp:1]: Unrelated change that made it into the PR?

Comment (hamishknight) [reply lib/DriverTool/swift_indent_main.cpp:1]: It's something I happened to notice when changing ParserUnit, I could put it in a separate PR, but then I'd have to rebase one of them.

**Review by rintaro — APPROVED** [review]

**Comment by hamishknight** [comment]
https://github.com/swiftlang/llvm-project/pull/9439

@swift-ci please test

**Comment by hamishknight** [comment]
https://github.com/swiftlang/llvm-project/pull/9439

@swift-ci please test Windows

**Comment by hamishknight** [comment]
https://github.com/swiftlang/llvm-project/pull/9439

@swift-ci please test

**Comment by hamishknight** [comment]
https://github.com/swiftlang/llvm-project/pull/9439

@swift-ci please test
